### PR TITLE
[mysqlreceiver] Use .my.cnf INI format for MySQL DBI password file

### PR DESCRIPTION
**Pull request by Klarsen04**
## Summary

Replace pgpass-style credential parsing with MySQL-native .my.cnf INI format for the MySQL DBI receiver password file.

## Changes

- **passfile.go**: Rewritten to parse INI `[section]` format instead of colon-separated pgpass lines
- **passfile_test.go**: Rewritten with 13 test cases for the new format

## Password file format

```ini
[client]
host=localhost
port=3306
user=cw_monitor
password=secret
```

Multi-instance:
```ini
[mysql_primary]
host=localhost
port=3306
user=cw_monitor
password=pass1

[mysql_secondary]
host=127.0.0.1
port=3307
user=cw_monitor
password=pass2
```

## Testing

- 13 unit tests covering: exact match, multi-instance, wrong user/host/port, missing fields, quotes, special chars, comments
- Verified on EC2 with 2 MySQL instances (3306 + 3307) — zero errors

### CODE REVIEW

File: receiver/mysqlreceiver/passfile.go [hunk 1]
```diff
@@ -10,68 +10,112 @@ import (
...
 	defer f.Close()
 
+	var sections []myCnfSection
+	var current *myCnfSection
+
 	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
 		line := strings.TrimSpace(scanner.Text())
-		if line == "" || strings.HasPrefix(line, "#") {
+		if line == "" || strings.HasPrefix(line, "#") || strings.HasPrefix(line, ";") {
```
Comment (JayPolanco) [inline receiver/mysqlreceiver/passfile.go]: Would this make sense to just be a regex?

File: receiver/mysqlreceiver/passfile.go [hunk 2]
```diff
@@ -10,68 +10,112 @@ import (
...
+	}
+	return lookupByMatching(sections, hostname, port, username, path)
+}
+
+type myCnfSection struct {
+	name   string
+	fields map[string]string
+}
+
+func parseMyCnfFile(path string) ([]myCnfSection, error) {
```
Comment (JayPolanco) [inline receiver/mysqlreceiver/passfile.go:38]: Could we not use the built in go ini pacakge (gopkg.in/ini.v1)?

Something like: 

```
  import "gopkg.in/ini.v1"

  type MySQLConfig struct {
      Host     string `ini:"host"`
      Port     int    `ini:"port"`
      User     string `ini:"user"`
      Password string `ini:"password"`
  }

  cfg, err := ini.Load("mysql.conf")
  if err != nil {
      return fmt.Errorf("load: %w", err)
  }

  configs := make(map[string]MySQLConfig)
```

File: receiver/mysqlreceiver/passfile.go [hunk 3]
```diff
@@ -10,68 +10,112 @@ import (
...
 
-// resolvePasswordFromPassfile reads a pgpass-inspired credential file and resolves
-// the password for the given connection parameters. The file format is:
+// resolvePasswordFromPassfile reads a MySQL option file (.my.cnf format) and
+// resolves the password for the given connection parameters.
 //
-//	hostname:port:database:username:password
+// Lookup strategy:
+//  1. Iterate all sections, match by host+port+user fields.
+//  2. If no section has match fields, fall back to the [client] section.
```
Comment (JayPolanco) [inline receiver/mysqlreceiver/passfile.go]: I don't see this fallback logic

File: receiver/mysqlreceiver/passfile.go [hunk 4]
```diff
@@ -10,68 +10,112 @@ import (
...
+	value := strings.TrimSpace(line[idx+1:])
+	value = stripINIQuotes(value)
+	return key, value, true
 }
 
-func matchField(pattern, value string) bool {
-	return pattern == "*" || pattern == value
+func stripINIQuotes(s string) string {
+	if len(s) >= 2 {
+		if (s[0] == '"' && s[len(s)-1] == '"') || (s[0] == 0x27 && s[len(s)-1] == 0x27) {
```
Comment (JayPolanco) [inline receiver/mysqlreceiver/passfile.go]: Nit: IIRC this is just the '\' char. Just looks a bit nicer

File: receiver/mysqlreceiver/passfile_test.go [hunk 5]
```diff
@@ -12,112 +12,156 @@ import (
...
 	dir := t.TempDir()
-	path := filepath.Join(dir, ".mysql_credentials")
-	require.NoError(t, os.WriteFile(path, []byte(content), 0o600))
+	path := filepath.Join(dir, ".mysql_password")
+	err := os.WriteFile(path, []byte(content), 0o600)
+	require.NoError(t, err)
+	return path
+}
 
+func TestResolvePasswordFromPassfile(t *testing.T) {
```
Comment (JayPolanco) [inline receiver/mysqlreceiver/passfile_test.go:24]: Don't think i saw a missing password test

File: receiver/mysqlreceiver/passfile.go (hunk 3 above)
Comment (Klarsen04) [reply receiver/mysqlreceiver/passfile.go]: You're right , that was a stale comment from an earlier design. Removed the fallback claim. The behavior is explicit match only (host+port+user must all match).

File: receiver/mysqlreceiver/passfile.go (hunk 2 above)
Comment (Klarsen04) [reply receiver/mysqlreceiver/passfile.go:38]: Good call. switched to gopkg.in/ini.v1.

File: receiver/mysqlreceiver/passfile.go (hunk 4 above)
Comment (Klarsen04) [reply receiver/mysqlreceiver/passfile.go]: thanks, will be deleting it, will be switching to use gopkg.in/ini.v1

File: receiver/mysqlreceiver/passfile.go (hunk 1 above)
Comment (Klarsen04) [reply receiver/mysqlreceiver/passfile.go]: the ini library handles section (switched to use this dependency gopkg.in/ini.v1)

File: receiver/mysqlreceiver/passfile_test.go (hunk 5 above)
Comment (Klarsen04) [reply receiver/mysqlreceiver/passfile_test.go:24]: Added. Thanks

File: receiver/mysqlreceiver/passfile.go [hunk 6]
```diff
@@ -4,74 +4,80 @@
...
 
-func matchField(pattern, value string) bool {
-	return pattern == "*" || pattern == value
+// matchField compares a field value against expected.
+// Both fields must be present and match (case-insensitive).
+func matchField(field, expected string) bool {
+	if field == "" || expected == "" {
+		return false
+	}
+	return strings.EqualFold(field, expected)
```
Comment (JayPolanco) [inline receiver/mysqlreceiver/passfile.go:82]: Are fields case sensitive?

File: receiver/mysqlreceiver/client_test.go
Comment (JayPolanco) [inline receiver/mysqlreceiver/client_test.go:154]: Lets make sure to update all the tests accordingly and also include tests for if username and password are empty

File: receiver/mysqlreceiver/passfile.go (hunk 6 above)
Comment (Klarsen04) [reply receiver/mysqlreceiver/passfile.go:82]: No, matching is case-insensitive. I use strings.EqualFold so Host=LOCALHOST matches localhost. There's a test case covering this (case insensitive matching).
